0.9.4 pre-release tasks

duncandewhurst opened the pull request:
**Related issues**

* Closes #423 

**Description**

https://github.com/open-contracting/infrastructure/pull/459/commits/512eb18d4d23dc6760470a3f9831dd5a2e4b6d19 adds to `manage.py` the script that I used to link field names to the schema reference tables. I thought that it might come in handy as it could be run against other Markdown content in the standard, but I'm happy to remove it if it's not going to be useful or if it's too much work to get the code up to the necessary standard.

**Merge checklist**

<!-- Complete the checklist before requesting a review. -->

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [x] `docs/examples/example.json`
  - [x] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [ ] Add an entry to `docs/reference/codelists.md`

duncandewhurst committed:
mapping/sustainability.yaml: Update elements, fix examples, lint

duncandewhurst committed:
mapping/sustainability.yaml: Fix life-cycle cost example

duncandewhurst committed:
manage.py (lint): Don't report missing examples for cross-references

duncandewhurst committed:
project-schema: Add missing Milestone.value from OCDS 1.2

duncandewhurst committed:
manage.py: Update infra_codelists and infra_definitions

duncandewhurst committed:
docs/reference/schema.md: Collapse references in project table

duncandewhurst committed:
docs/reference/codelists.md: Remove duplicate heading, fix ordering

duncandewhurst committed:
docs/cost/ids/sustainability.md: Add reminder to look at schema

duncandewhurst committed:
docs/cost/common.md: Copy edit, reference `.pageStart` and .pageEnd`

duncandewhurst committed:
mapping/sustainability.yaml: Copy-edit mappings

duncandewhurst committed:
mapping/sustainability.yaml: Add links to sub-schema reference

Regex find: `([a-zA-Z.]*)` object
Regex replace: [`$1`](../../reference/schema.md#\l$1) object

duncandewhurst committed:
mapping/sustainability.yaml: Link field names to schema reference

duncandewhurst committed:
manage.py (lint): Link field names to jsonschema directives

duncandewhurst committed:
Merge branch '0.9-dev' into 423-pre-release-tasks

duncandewhurst committed:
docs/cost/ids/sustainability.md: Run update-sustainability-docs

duncandewhurst committed:
Merge branch '0.9-dev' into 423-pre-release-tasks

duncandewhurst committed:
project-schema: Add minProperties to ClimateMeasure

duncandewhurst committed:
mapping/sustainability.yaml: Update climate measures and protected area

duncandewhurst committed:
schema/project-level/project-schema.json: Reorder top-level fields

duncandewhurst committed:
Update changelog

duncandewhurst committed:
docs/examples/blank.json: Add Milestone.value

duncandewhurst commented:
@odscjen for the changes to `sustainability.yaml` and `sustainability.md`, I think the best thing to do would be to review the [HTML version](https://standard.open-contracting.org/staging/infrastructure/423-pre-release-tasks/en/cost/ids/sustainability/). For the changes to `project-schema.json`, best to review https://github.com/open-contracting/infrastructure/pull/459/commits/4fbfb325c8a636424322e041066b9db6bbd24c91 separately and just check that the order matches that specified in #423.

duncandewhurst committed:
Remove erroneously committed file

odscjen commented on docs/cost/common.md:
```diff
@@ -4,24 +4,26 @@ To avoid repetition in the mapping, we refer and link to the following common op
+* `.id` incrementally
+* `.url` to a direct link to the document
+* `.title` to the title of the document
+* `.pageStart` and `pageEnd` to the page range containing the information described in its `.documentType`
```
```suggestion
* `.pageStart` and `.pageEnd` to the page range containing the information described in its `.documentType`
```

odscjen commented on docs/cost/common.md:
```diff
@@ -4,24 +4,26 @@ To avoid repetition in the mapping, we refer and link to the following common op
+* `.id` incrementally
+* `.url` to a direct link to the document
+* `.title` to the title of the document
+* `.pageStart` and `pageEnd` to the page range containing the information described in its `.documentType`
```
```suggestion
* `.pageStart` and `.pageEnd` to the page range containing the information described in its `.documentType`
```

odscjen commented on docs/cost/common.md:
```diff
@@ -4,24 +4,26 @@ To avoid repetition in the mapping, we refer and link to the following common op
-* Add an `Organization` object to the `.parties` array and set its `.name` to the name of the organization.
-* If you collect organization identifiers, set `.identifier` according to the [identifier reference](../reference/schema.md#identifier) and set `.id` to {identifier.scheme}-{identifier.id}(-{department-identifier}). Otherwise, set `.id` incrementally.
+1. Add an `Organization` object to the `.parties` array and set its `.name` to the name of the organization.
+1. If you collect organization identifiers, set `.identifier` according to the [identifier reference](../reference/schema.md#identifier) and set `.id` to {identifier.scheme}-{identifier.id}(-{department-identifier}). Otherwise, set `.id` incrementally.
```
Not sure how to fix this but it's interpreting "{identifier.id}(-{department-identifier})" as an html link, but obviously it's not so doesn't go anywhere. Possibly use square brackets about the department-identifier rather than round ones? I'm assuming it's in brackets to indicate that the department-identifier may not be relevant.

duncandewhurst replied:
Fixed in a0a93f3881f81d37a8bc76be9e30df9018a44376

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -289,10 +284,10 @@
-        - Set `.environment.hasImpactAssessment` to `true`.
-        - Add a `Classification` object to the `.environment.impactCategories` array, set its `.scheme` to "ifc-environmental-social" and set its `.id` to the letter for the category into which the project falls.
+        1. Set [`.environment.hasImpactAssessment`](project-schema.json,,environment/hasImpactAssessment) to `true`.
+        2. Add a [`Classification`](../../reference/schema.md#classification) object to the [`.environment.impactCategories`](project-schema.json,,environment/impactCategories) array, set its [`.scheme`](project-schema.json,/definitions/Classification,scheme) to "ifc-environmental-social" and set its [`.id`](project-schema.json,/definitions/Classification,id) to the lowercase letter of the catego ...
```
```suggestion
        2. Add a [`Classification`](../../reference/schema.md#classification) object to the [`environment.impactCategories`](project-schema.json,,environment/impactCategories) array, set its [`.scheme`](project-schema.json,/definitions/Classification,scheme) to 'ifc-environmental-social' and set its [`.id`](project-schema.json,/definitions/Classification,id) to the lowercase letter of the category into which the project falls.
```

The other codes so far have all used single quotes with double quotes being used for set text so I'm just going with that convention now

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -572,19 +569,16 @@
 
-        - `.id` incrementally
-        - `.date` to the date that the forecast was prepared.
+        2. Add a [`CostGroup`](../../reference/schema.md#costgroup) object to the cost measurement's [`.costGroups`](project-schema.json,/definitions/CostMeasurement,costGroups) array, set its [`.id`](project-schema.json,/definitions/CostGroup,id) incrementally and set its [`.category`](project-schema.json,/definitions/CostGroup,category) to 'endOfLife'
```
```suggestion
        2. Add a [`CostGroup`](../../reference/schema.md#costgroup) object to the cost measurement's [`.costGroups`](project-schema.json,/definitions/CostMeasurement,costGroups) array, set its [`.id`](project-schema.json,/definitions/CostGroup,id) incrementally and set its [`.category`](project-schema.json,/definitions/CostGroup,category) to 'endOfLife'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -707,7 +700,7 @@
     module: Climate finance
     indicator: Policy coherence
-    disclosure format: Select from a list of international commitments and clarify how the project is aligned with it \[codelist, free text to explain\].
+    disclosure format: Clarify how the investment is aligned with the country's nationally determined contributions. \[codelist, free text to explain\].
```
do we need the example here? there's no example for the other policy alignment data points in this module and there's an example in the linked to data point in the institutional module?

duncandewhurst replied:
Removed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -289,10 +284,10 @@
 
-        - Set `.environment.hasImpactAssessment` to `true`.
-        - Add a `Classification` object to the `.environment.impactCategories` array, set its `.scheme` to "ifc-environmental-social" and set its `.id` to the letter for the category into which the project falls.
+        1. Set [`.environment.hasImpactAssessment`](project-schema.json,,environment/hasImpactAssessment) to `true`.
```
```suggestion
        1. Set [`environment.hasImpactAssessment`](project-schema.json,,environment/hasImpactAssessment) to `true`.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -348,11 +343,11 @@
-        1. Set `.environment.inProtectedArea` to `true`.
-        2. Add a `Location` object to the `.locations` array, set its `.id` incrementally and set its `.gazetteer.scheme` to 'WDPA'
-        3. For each protected area in which the project is located, add the area's WDPA identifier to the `.gazetteer.identifiers` array.
+        1. Set [`.environment.inProtectedArea`](project-schema.json,,environment/inProtectedArea) to `true`.
```
```suggestion
        1. Set [`environment.inProtectedArea`](project-schema.json,,environment/inProtectedArea) to `true`.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -348,11 +343,11 @@
+        3. For each protected area in which the project is located, add the area's WDPA identifier to the [`.gazetteer.identifiers`](project-schema.json,/definitions/Location,gazetteer/identifiers) array.
 
-        If the project is not located in a protected area, set `.environment.inProtectedArea` to `false`.
+        If the project is not located in a protected area, set [`.environment.inProtectedArea`](project-schema.json,,environment/inProtectedArea) to `false`.
```
```suggestion
        If the project is not located in a protected area, set [`environment.inProtectedArea`](project-schema.json,,environment/inProtectedArea) to `false`.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1153,26 +1147,23 @@
-        Project level:
-
-        For each KPI metric add the sentence "To be reported XXX" where XXX is the reporting frequency to the metrics `.description`.
+    mapping: 'Project level: Get the [`Metric`](../../reference/schema.md#metric) object in the [`.metrics`](project-schema.json,,metrics) array that represents the KPI and map to its [`.description`](project-schema.json,/definitions/Metric,description) using the format "Reporting period: <period>".'
```
```suggestion
    mapping: |
        Project level: 
        
        Get the [`Metric`](../../reference/schema.md#metric) object in the [`metrics`](project-schema.json,,metrics) array that represents the KPI and map to its [`.description`](project-schema.json,/definitions/Metric,description) using the format "Reporting period: {period}".
```

"Reporting period: <period>" is becoming "Reporting period: " in the rendered page so need to use something else around the word to ensure it shows up

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1195,7 +1186,7 @@
     indicator: Long-term project impact
     disclosure format: Disclose technical audits produced at end of the project (E.g. \[free text\].
-    mapping: 'Project-level: For each audit report [add a project document](../common.md#add-a-project-document), set `.documentType` to ''technicalAuditReport'' and map any free text to `.description` .'
+    mapping: 'Project-level: For each audit report, [add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''technicalAuditReport'' and map any free text to its [`.description`](project-schema.json,/definitions/Document,description) .'
```
```suggestion
    mapping: |
        Project level: 
        
        For each audit report, [add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'technicalAuditReport' and map any free text to its [`.description`](project-schema.json,/definitions/Document,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1285,15 +1276,15 @@
-        - `.relationship` to 'replacement'
+        - [`.id`](project-schema.json,/definitions/RelatedProject,id) and [`.title`](project-schema.json,/definitions/RelatedProject,title) to the [`.id`](project-schema.json,,id) and [`.title`](project-schema.json,,title) of the OC4IDS project for the replacement of the asset
+        - [`.scheme`](project-schema.json,/definitions/RelatedProject,scheme) to 'oc4ids'
+        - [`.relationship`](project-schema.json,/definitions/RelatedProject,relationship) to 'replacement'
```
```suggestion
        - [`.relationship`](project-schema.json,/definitions/RelatedProject,relationship) to 'replacement'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1468,10 +1456,10 @@
 
-        1. Set `.social.inIndigenousLand` to `true`
-        2. Add a `Location` object to the `.locations` array, set its `.id` incrementally and set its description to "Indigenous land: <Name> (<Category>)" substituting <Name>  and <Category> for the name and land category from the Landmark database.
+        1. Set [`.social.inIndigenousLand`](project-schema.json,/definitions/Social,inIndigenousLand) to `true`
```
```suggestion
        1. Set [`social.inIndigenousLand`](project-schema.json,/definitions/Social,inIndigenousLand) to `true`.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1468,10 +1456,10 @@
+        2. Add a [`Location`](../../reference/schema.md#location) object to the [`.locations`](project-schema.json,,locations) array, set its [`.id`](project-schema.json,/definitions/Location,id) incrementally and set its description to "Indigenous land: <Name> (<Category>)" substituting <Name> and <Category> for the name and land category from the Landmark database.
 
-        If the project is not located in or cutting through indigenous land, set `.social.inIndigenousLand` to `false`.
+        If the project is not located in and does not cut through indigenous land, set [`.social.inIndigenousLand`](project-schema.json,/definitions/Social,inIndigenousLand) to `false`.
```
```suggestion
        If the project is not located in and does not cut through indigenous land, set [`social.inIndigenousLand`](project-schema.json,/definitions/Social,inIndigenousLand) to `false`.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1494,16 +1482,16 @@
+        1. Publish the meeting invite: [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'consultationMeetingInvitation'.
 
-        2. Publish the meeting details. Add a `Meeting` object to the `.social.consultationMeetings` array and set:
+        2. Publish the meeting details: add a [`Meeting`](../../reference/schema.md#meeting) object to the [`.social.consultationMeetings`](project-schema.json,/definitions/Social,consultationMeetings) array and set its:
```
```suggestion
        2. Publish the meeting details: add a [`Meeting`](../../reference/schema.md#meeting) object to the [`social.consultationMeetings`](project-schema.json,/definitions/Social,consultationMeetings) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1540,7 +1528,7 @@
     indicator: Land issues
     disclosure format: Disclose budget allocated to fund land compensation (E.g. \[value\]).
-    mapping: 'Project Level: Set `.social.landCompensationBudget` to the amount and currency of the budget allocated for land compensation.'
+    mapping: 'Project Level: Map to [`.social.landCompensationBudget`](project-schema.json,/definitions/Social,landCompensationBudget).'
```
```suggestion
    mapping: |
        Project Level: 
        
        Map to [`social.landCompensationBudget`](project-schema.json,/definitions/Social,landCompensationBudget).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1494,16 +1482,16 @@
+        - [`.id`](project-schema.json,/definitions/Meeting,id) incrementally
+        - [`.date`](project-schema.json,/definitions/Meeting,date) to the date of the meeting
+        - [`.address`](project-schema.json,/definitions/Meeting,address) to the address of the meeting
+        - [`.numberOfParticipants`](project-schema.json,/definitions/Meeting,numberOfParticipants) to the number of people that participated in the meeting
```
```suggestion
        - [`.numberOfParticipants`](project-schema.json,/definitions/Meeting,numberOfParticipants) to the number of people that participated in the meeting.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1576,14 +1564,14 @@
 
-        1. For each labor obligation in the contract, add a code from the [laborObligations](../../reference/codelists.md#laborobligations) codelist to the`.summary.social.laborObligations.obligations` array.
-        2. Optionally, add a further explanation of the labor obligations to `.summary.social.laborObligations.description`.
+           1. For each labor obligation, add a code from the open [laborObligations](../../reference/codelists.md#laborobligations) codelist to the [`.summary.social.laborObligations.obligations`](project-schema.json,/definitions/ContractingProcessSummary,social/laborObligations/obligations) array.
```
```suggestion
           1. For each labor obligation, add the relevant code from the open [laborObligations](../../reference/codelists.md#laborobligations) codelist to the [`summary.social.laborObligations.obligations`](project-schema.json,/definitions/ContractingProcessSummary,social/laborObligations/obligations) array.
```

"add *a* code..." feels a bit ambiguous, I don't think this was picked up in the PR that created this initially.

duncandewhurst replied:
Done in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1647,13 +1635,27 @@
+        - [`.period`](project-schema.json,/definitions/Observation,period) to the reporting period
+        - [`.measure`](project-schema.json,/definitions/Observation,measure) to the number of accidents.
+
+        3. If further details available, add them to the observation's [`.notes`](project-schema.json,/definitions/Observation,notes) field.
```
```suggestion
        3. If further details are available, add them to the observation's [`.notes`](project-schema.json,/definitions/Observation,notes) field.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -3,7 +3,7 @@
     indicator: Procurement viability
     disclosure format: Disclose the procurement strategy risk assessment. This tends to be part of the decision-making strategy and likely includes discussions regarding capabilities, the delivery model and the rationale for the risk allocation decision. (E.g. \[Document\]).
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''procurementStrategyRiskAssessment''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''procurementStrategyRiskAssessment''.'
```
```suggestion
    mapping: | 
        Project Level: 
    
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'procurementStrategyRiskAssessment'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -20,15 +20,14 @@
     indicator: Economic viability
     disclosure format: Disclose the life cycle cost of the project, which is the cost of an asset throughout its life cycle while fulfilling the performance requirements (ISO 15686-5:2017) (E.g. \[value\]).
-    mapping: '**Project Level:** Add a `CostMeasurement` object to the `costMeasurements` array. Set `.date` to the date the analysis was prepared. Map to the cost measurement''s `.lifeCycleCost.cost`.'
+    mapping: 'Project Level: Add a [`CostMeasurement`](../../reference/schema.md#costmeasurement) object to the [`costMeasurements`](project-schema.json,,costMeasurements) array and map to its [`.lifeCycleCosting.value`](project-schema.json,/definitions/CostMeasurement,lifeCycleCosting/value).'
```
```suggestion
    mapping: | 
        Project Level: 
        
        Add a [`CostMeasurement`](../../reference/schema.md#costmeasurement) object to the [`costMeasurements`](project-schema.json,,costMeasurements) array and map to its [`.lifeCycleCosting.value`](project-schema.json,/definitions/CostMeasurement,lifeCycleCosting/value).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -41,7 +40,7 @@
     indicator: Economic viability
     disclosure format: Disclose the methodology used to calculate the life-cycle cost. The methodology ought to specify whether income and externalities are included in the calculation and the common date, discount rate and period of analysis used.
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''lifeCycleCostMethodology''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''lifeCycleCostMethodology''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'lifeCycleCostMethodology'.
```
Also I can't select it for a suggestion as it's not a change from this PR but noticed that the `documentType` is "lifecycleCostMethodology" in the accompanying example, but it should be "lifeCycleCostMethodology"

duncandewhurst replied:
Fixed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -156,14 +155,14 @@
     indicator: Economic viability
     disclosure format: Disclose the project cost-benefit analysis. This information tends to be part of the appraisal documents. (E.g. \[Document\]).
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''costBenefitAnalysis''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''costBenefitAnalysis''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'costBenefitAnalysis'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -172,7 +171,7 @@
     indicator: Economic viability
     disclosure format: A summary of the value for money analysis carried out for the project, along with supporting figures, calculations and business case, based on projected or actual procurement outcomes. This tends to include considerations of economy, efficiency, effectiveness and equity, and is part of the appraisal documents. (E.g. \[Document\]).
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''valueForMoneyAnalysis''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''valueForMoneyAnalysis''.'
```
```suggestion
    mapping: |
        
        Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'valueForMoneyAnalysis'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -310,7 +305,7 @@
     indicator: Environmental impact
     disclosure format: Disclose the measures adopted by the project to mitigate and/or remedy the environmental impact (E.g. \[free text justification/explaining the measures adopted\] and document).
-    mapping: 'Project level: Add a document, set its `.documentType` to ''environmentalMeasures'', add a description of the environmental measures adopted to its `.description` and set  `.url` to the URL of the document that describes the project''s environmental measures (if available).'
+    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''environmentalMeasures'' and map the justification and explanation to its [`.description`](project-schema.json,/definitions/Document,description).'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'environmentalMeasures' and map the justification and explanation to its [`.description`](project-schema.json,/definitions/Document,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -327,7 +322,7 @@
     indicator: Environmental impact
     disclosure format: Disclose all exemptions and/or amnesties obtained for the project. (E.g. \[Document\]). This can be planning, environmental, construction and/or operational related.
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''environmentalExemption''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''environmentalExemption''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'environmentalExemption'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -530,7 +527,7 @@
     indicator: Climate management
     disclosure format: Disclose environmental and/or climate related certifications issued for contractors and subcontractors such as ISO 14001 for environmental management (E.g. \[Document\]).
-    mapping: 'Contracting process level: [Add a contracting process document](../common.md#add-a-contracting-process-document) and set `.documentType` to ''supplierEnvironmentalCertification''.'
+    mapping: 'Contracting process level: [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''supplierEnvironmentalCertification''.'
```
```suggestion
    mapping: | 
        Contracting process level: 
        
        [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'supplierEnvironmentalCertification'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -553,7 +550,7 @@
     indicator: Climate management
     disclosure format: Disclose the decommissioning plans for the project assets \[document\]..
-    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''decommissioningPlans''.'
+    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''decommissioningPlans''.'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'decommissioningPlans'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -616,12 +610,13 @@
         - adaptation
         - cross-cutting
-    mapping: 'Project level: Add the relevant codes from the environmentalGoal codelist to the `environment.goals` array'
+    mapping: 'Project level: Map to [`environment.goals`](project-schema.json,,environment/goals) using the closed [environmentalGoal](../../reference/codelists.md#environmentalgoal) codelist.'
```
```suggestion
    mapping: |
        Project level: 
        
        Map to [`environment.goals`](project-schema.json,,environment/goals) using the closed [environmentalGoal](../../reference/codelists.md#environmentalgoal) codelist.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -672,16 +667,14 @@
     indicator: Model of change
     disclosure format: Clarify the theory of change, systemic transition or transformation that is intended \[free text to explain\].
-    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document), set `.documentType` to ''climateTransformation'' and map any free text to `.description`.'
+    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''climateTransformation'' and map any free text to [`.description`](project-schema.json,/definitions/Document,description).'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'climateTransformation' and map any free text to [`.description`](project-schema.json,/definitions/Document,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -690,7 +683,7 @@
     indicator: Decision-making process
     disclosure format: Identify who approved the climate finance investment in the country (organization, party, role)
-    mapping: 'Project level: [Add an organization](../common.md#add-an-organization) for the decision maker and add ''climateFinanceFocalPoint'' to its `.roles` array.'
+    mapping: 'Project level: [Add an organization](../common.md#add-an-organization) for the decision maker and add ''climateFinanceFocalPoint'' to its [`.roles`](project-schema.json,/definitions/Organization,roles) array.'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add an organization](../common.md#add-an-organization) for the decision maker and add 'climateFinanceFocalPoint' to its [`.roles`](project-schema.json,/definitions/Organization,roles) array.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -987,7 +982,7 @@
     indicator: Climate impact
     disclosure format: Disclose the cost per tonne of CO2 equivalent \[value, currency\].
-    mapping: Map to `environment.abatementCost`. If supporting documentation is available, [add a project document](../common.md#add-a-project-document) and set `.documentType` to 'abatementCostMethodology'.
+    mapping: 'Project level: Map to [`environment.abatementCost`](project-schema.json,,environment/abatementCost). If a document that describes the methodology used to calculate the cost is available, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''abatementCostMethodology''.'
```
```suggestion
    mapping: |
        Project level: 
        
        Map to [`environment.abatementCost`](project-schema.json,,environment/abatementCost). If a document that describes the methodology used to calculate the cost is available, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'abatementCostMethodology'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1015,7 +1010,7 @@
         gender empowerment
         \[add free text to explain the co-benefits\]
-    mapping: 'Project level: For each co-benefit, add a `Benefit` object to the `benefits` array, map the option from the list to its `.title` and map the explanation to its `.description`.'
+    mapping: 'Project level: For each co-benefit, add a [`Benefit`](../../reference/schema.md#benefit) object to the [`benefits`](project-schema.json,,benefits) array, map the option from the list to its [`.title`](project-schema.json,/definitions/Benefit,title) and map the explanation to its [`.description`](project-schema.json,/definitions/Benefit,description).'
```
```suggestion
    mapping: |
        Project level: 
        
        For each co-benefit, add a [`Benefit`](../../reference/schema.md#benefit) object to the [`benefits`](project-schema.json,,benefits) array, map the option from the list to its [`.title`](project-schema.json,/definitions/Benefit,title) and map the explanation to its [`.description`](project-schema.json,/definitions/Benefit,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1121,12 +1117,13 @@
         external
         mixed
-    mapping: 'Project level: Add the relevant codes from the climateOversightTypes codelist to the `environment.climateOversightTypes` array'
+    mapping: 'Project level: Map to [`environment.climateOversightTypes`](project-schema.json,,environment/climateOversightTypes) using the closed [climateOversightTypes](../../reference/codelists.md#climateoversighttypes) codelist.'
```
```suggestion
    mapping: |
       Project level: 
       
       Map to [`environment.climateOversightTypes`](project-schema.json,,environment/climateOversightTypes) using the closed [climateOversightTypes](../../reference/codelists.md#climateoversighttypes) codelist.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1135,10 +1132,7 @@
-        Project level:
-
-        For each KPI add a `Metric` object to the `metrics` array and set the object's fields according to the schema. Prefix the metric's `.title` with "KPI".
+    mapping: 'Project level: For each KPI, add a [`Metric`](../../reference/schema.md#metric) object to the [`.metrics`](project-schema.json,,metrics) array and set its fields according to the schema. Prefix the metric''s [`.title`](project-schema.json,/definitions/Metric,title) with "KPI: ".'
```
```suggestion
    mapping: |
        Project level: 
        
        For each KPI, add a [`Metric`](../../reference/schema.md#metric) object to the [`metrics`](project-schema.json,,metrics) array and set its fields according to the schema. Prefix the metric's [`.title`](project-schema.json,/definitions/Metric,title) with "KPI: ".
```

moved this back just so that everything has the same look

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1153,26 +1147,23 @@
-    disclosure format: \[document\]
-    mapping: 'Project-level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''oversightReport''.'
+    disclosure format: Disclose oversight reports.\[document\]
+    mapping: 'Project-level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''oversightReport''.'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'oversightReport'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1214,7 +1205,7 @@
     indicator: Long-term project impact
     disclosure format: Clarify the methodology or system to measure the long-term impact of the project solution (E.g. \[free text\].
-    mapping: 'Project-level: [Add a project document](../common.md#add-a-project-document), set `.documentType` to ''impactMethodology''and map any free text to `.description`.'
+    mapping: 'Project-level: [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''impactMethodology'' and map any free text to its [`.description`](project-schema.json,,description).'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'impactMethodology' and map any free text to its [`.description`](project-schema.json,,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1313,7 +1304,7 @@
     indicator: Stranded assets
     disclosure format: Intended start and end dates of decommissioning.
-    mapping: Map to `decommissioningPeriod`.
+    mapping: Map to [`decommissioningPeriod`](project-schema.json,,decommissioningPeriod).
```
```suggestion
    mapping: |
        Project level:
        
        Map to [`decommissioningPeriod`](project-schema.json,,decommissioningPeriod).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1385,10 +1376,7 @@
-        Project level:
-
-        Publish in documents, with `.documentType` set to 'socialImpact' and include a short description and/or a link to a document providing details. If the details are part of a more general document indicate the relevant section of the document using `.pageStart` and `.pageEnd`.
+    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''socialImpact''.'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'socialImpact'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1620,7 +1608,7 @@
     indicator: Labour compliance
     disclosure format: Disclose the amount allocated by the main contractor to cover for labour costs (E.g. \[value\]).
-    mapping: 'Contracting process level: Map to `.summary.social.laborBudget`.'
+    mapping: 'Contracting process level: Map to [`.summary.social.laborBudget`](project-schema.json,/definitions/ContractingProcessSummary,social/laborBudget).'
```
```suggestion
    mapping: |
        Contracting process level: 
        
        Map to [`summary.social.laborBudget`](project-schema.json,/definitions/ContractingProcessSummary,social/laborBudget).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1693,7 +1695,7 @@
     indicator: Health and Safety
     disclosure format: Disclose labour related certifications issued in relation to project contractors and subcontractors such as ISO 45001 for Health and Safety (E.g. \[Document\]).
-    mapping: 'Contracting process level: [Add a contracting process document](../common.md#add-a-contracting-process-document) and set `.documentType` to ''supplierHealthAndSafetyCertification''.'
+    mapping: 'Contracting process level: [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''supplierHealthAndSafetyCertification''.'
```
```suggestion
    mapping: |
        Contracting process level: 
        
        [Add a contracting process document](../common.md#add-a-contracting-process-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'supplierHealthAndSafetyCertification'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1770,7 +1775,7 @@
     indicator: Health and Safety
     disclosure format: Disclose building inspections during project implementation (E.g. \[Document\]).
-    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set `.documentType` to ''buildingInspectionReport''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''buildingInspectionReport''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'buildingInspectionReport'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1791,27 +1796,25 @@
 
-        1. Add a `Metric` object to the `forecasts` array and set its `.title` to 'Jobs created'.
-        2. Add an `observation` object for each job relationship - stage pair and set `.dimensions` accordingly, i.e.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`forecasts`](project-schema.json,,forecasts) array and set its [`.title`](project-schema.json,/definitions/Metric,title) to 'Jobs created'.
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`forecasts`](project-schema.json,,forecasts) array and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Jobs created".
```
set text rather than a code so double quotes

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1791,27 +1796,25 @@
 
-        3. Record the estimated number of jobs created for each relationship and stage in the relevant observation's `.measure`.
+        - [`.id`](project-schema.json,/definitions/Observation,id) incrementally
+        - [`.dimensions.relationship`](project-schema.json,/definitions/Observation,dimensions) to the relationship ("direct" or "indirect")
```
neither `dimensions.relationship` not `dimensions.stage` exist!

duncandewhurst replied:
`dimensions` permits any key/value pairs so that's OK :-)

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1916,13 +1919,21 @@
         Project level:
 
-        For each plan or policy to which the project is aligned, add a code from the [policyAlignment](../../reference/codelists.md#policyalignment) codelist to the `.policyAlignment.policies` array. Add a further explanation of the project's policy alignment to `.policyAlignment.description`.
+        1. For each plan or policy to which the project is aligned, add a code from the open [policyAlignment](../../reference/codelists.md#policyalignment) codelist to the [`.policyAlignment.policies`](project-schema.json,,policyAlignment/policies) array and add a further explanation of the alignment to [`.policyAlignment.description`](project-schema.json,,policyAlignment/description).
```
```suggestion
        1. For each plan or policy to which the project is aligned, add a code from the open [policyAlignment](../../reference/codelists.md#policyalignment) codelist to the [`policyAlignment.policies`](project-schema.json,,policyAlignment/policies) array and add a further explanation of the alignment to [`policyAlignment.description`](project-schema.json,,policyAlignment/description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1916,13 +1919,21 @@
-        For each plan or policy to which the project is aligned, add a code from the [policyAlignment](../../reference/codelists.md#policyalignment) codelist to the `.policyAlignment.policies` array. Add a further explanation of the project's policy alignment to `.policyAlignment.description`.
+        1. For each plan or policy to which the project is aligned, add a code from the open [policyAlignment](../../reference/codelists.md#policyalignment) codelist to the [`.policyAlignment.policies`](project-schema.json,,policyAlignment/policies) array and add a further explanation of the alignment to [`.policyAlignment.description`](project-schema.json,,policyAlignment/description).
+
+        2. For each United Nations Sustainable Development Goal to which the project is aligned, add a [`Classification`](../../reference/schema.md#classification) object to the [`.additionalClassifications`](project-schema.json,,additionalClassifications) array and set its:
```
```suggestion
        2. For each United Nations Sustainable Development Goal to which the project is aligned, add a [`Classification`](../../reference/schema.md#classification) object to the [`additionalClassifications`](project-schema.json,,additionalClassifications) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1916,13 +1919,21 @@
+        - [`.description`](project-schema.json,/definitions/Classification,description) to the goal's title
 
-        For each United Nations Sustainable Development Goal to which the project is aligned, add a `Classification` object to the `.additionalClassifications` array, set its `.scheme` to 'sdg', set its `.id` to the goal's number and set its `.description` to the goal's title.
+        3. For each United Nations Sustainable Development Goal Target to which the project is aligned, add a [`Classification`](../../reference/schema.md#classification) object to the [`.additionalClassifications`](project-schema.json,,additionalClassifications) array and set its:
```
```suggestion
        3. For each United Nations Sustainable Development Goal Target to which the project is aligned, add a [`Classification`](../../reference/schema.md#classification) object to the [`additionalClassifications`](project-schema.json,,additionalClassifications) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1956,10 +1967,7 @@
-        Project level:
-
-        For each freedom of information request, [add a project document](../common.md#add-a-project-document) and set `.documentType` to 'informationRequest'.
+    mapping: 'Project level: For each freedom of information request, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''informationRequest''.'
```
```suggestion
    mapping: |
        Project level: 
        
        For each freedom of information request, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'informationRequest'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1975,10 +1983,7 @@
-        Project level:
-
-        For each freedom of information request response, [add a project document](../common.md#add-a-project-document) and set `.documentType` to 'informationRequestResponse'.
+    mapping: 'Project level: For each freedom of information request response, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''informationRequestResponse''.'
```
```suggestion
    mapping: |
        Project level: 
        
        For each freedom of information request response, [add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'informationRequestResponse'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1999,17 +2004,22 @@
-        5. Publish the meeting minutes. [Add a project document](../common.md#add-a-project-document) and set its `.documentType` to 'minutes.lobbyingMeeting'.
+        Publish the meeting details:
+
+        1. Add a [`Meeting`](../../reference/schema.md#meeting) object to the [`.lobbyingMeetings`](project-schema.json,,lobbyingMeetings) array and set:
```
```suggestion
        1. Add a [`Meeting`](../../reference/schema.md#meeting) object to the [`lobbyingMeetings`](project-schema.json,,lobbyingMeetings) array and set:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1999,17 +2004,22 @@
+        - [`.address`](project-schema.json,/definitions/Meeting,address) to the address of the meeting
+        - [`.numberOfParticipants`](project-schema.json,/definitions/Meeting,numberOfParticipants) to the number of people present at the meeting
+        - [`.publicOffice.person.name`](project-schema.json,/definitions/PublicOffice,person/name) to the name of the person representing the public office present at the meeting
+        - [`.publicOffice.jobTitle`](project-schema.json,/definitions/PublicOffice,jobTitle) to the job title of the person representing the public office present at the meeting
```
```suggestion
        - [`.publicOffice.jobTitle`](project-schema.json,/definitions/PublicOffice,jobTitle) to the job title of the person representing the public office present at the meeting.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2081,10 +2094,7 @@
-        Contracting process level:
-
-        Add a `Sustainability` object to the `summary.tender.sustainability` array and add 'awardCriteria' to its `.strategies` array.
+    mapping: 'Contracting process level: Add a [`Sustainability`](../../reference/schema.md#sustainability) object to the [`summary.tender.sustainability`](project-schema.json,/definitions/ContractingProcessSummary,tender/sustainability) array and add ''awardCriteria'' to its [`.strategies`](project-schema.json,/definitions/Sustainability,strategies) array.'
```
```suggestion
    mapping: |
        Contracting process level: 
        
        Add a [`Sustainability`](../../reference/schema.md#sustainability) object to the [`summary.tender.sustainability`](project-schema.json,/definitions/ContractingProcessSummary,tender/sustainability) array and add 'awardCriteria' to its [`.strategies`](project-schema.json,/definitions/Sustainability,strategies) array.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2109,7 +2119,7 @@
     indicator: Anti-corruption impact
     disclosure format: Disclose anti-corruption certifications of the project, such as ISO 37001 on Anti-Bribery Management Systems Standard (E.g. \[Document\]).
-    mapping: 'Project Level: [add a project document](../common.md#add-a-project-document) and set `.documentType` to ''antiCorruptionCertification''.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''antiCorruptionCertification''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'antiCorruptionCertification'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2125,7 +2135,7 @@
     indicator: Anti-corruption impact
     disclosure format: Identify the entities acting as independent monitors of the project (E.g. \[free text\])
-    mapping: 'Project level: [Add an organization](../common.md#add-an-organization) for the independent monitor and add ''independentMonitor'' to its `.roles` array.'
+    mapping: 'Project level: [Add an organization](../common.md#add-an-organization) for the independent monitor and add ''independentMonitor'' to its [`.roles`](project-schema.json,/definitions/Organization,roles) array.'
```
```suggestion
    mapping: |
       Project level: 
       
       [Add an organization](../common.md#add-an-organization) for the independent monitor and add 'independentMonitor' to its [`.roles`](project-schema.json,/definitions/Organization,roles) array.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2143,22 +2153,14 @@
     indicator: Risk management
     disclosure format: Disclose risk management plans prepared for the project (E.g. \[Document\]).
-    mapping: 'Project Level: Publish in documents, with .documentType set to ''riskProvisions'' and include a short description and/or a link to a document providing details. If the details are part of a more general document indicate the relevant section of the document using `.pageStart` and `.pageEnd`.'
+    mapping: 'Project Level: [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''riskProvisions''.'
```
```suggestion
    mapping: |
        Project Level: 
        
        [Add a project document](../common.md#add-a-project-document) and set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'riskProvisions'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2177,19 +2179,20 @@
 
         Free text to add not mentioned sub-sectors
-    mapping: 'Project Level: Map to `sector`, using the \[ProjectSector codelist\]((../../reference/codelists.md#projectsector).'
+    mapping: 'Project Level: Map to [`sector`](project-schema.json,,sector), using the open [ProjectSector codelist](../../reference/codelists.md#projectsector).'
```
```suggestion
    mapping: |
        Project Level: 
        
        Map to [`sector`](project-schema.json,,sector), using the open [ProjectSector codelist](../../reference/codelists.md#projectsector).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -60,15 +59,15 @@
+           a. Get the [`BudgetBreakdown`](../../reference/schema.md#budgetbreakdown) object in the budget breakdowns' [`.budgetBreakdown`](project-schema.json,/definitions/BudgetBreakdowns,budgetBreakdown) array whose [`.description`](project-schema.json,/definitions/BudgetBreakdown,description) is the name of the stage. If none exists yet, add a [`BudgetBreakdown`](../../reference/schema.md#budg ...
 
-           b. Get the `Organization` object in `.parties` that represents the funding source. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its `.roles`.
+           b. Get the [`Organization`](../../reference/schema.md#organization) object in [`.parties`](project-schema.json,,parties) that represents the funding source. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its [`.roles`](project-schema.json,/definitions/Organization,roles).
```
```suggestion
           b. Get the [`Organization`](../../reference/schema.md#organization) object in [`parties`](project-schema.json,,parties) that represents the funding source. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its [`.roles`](project-schema.json,/definitions/Organization,roles).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -241,12 +240,12 @@
-        - `.value` to the value of the shortfall
-        - `.notes` to the reasons for the shortfall
-        - optionally, `.period` to the period to which the shortfall applies
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Budget shortfall".
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Budget shortfall".
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -418,15 +413,15 @@
     indicator: Climate and disaster risk
     disclosure format: Clarify the type of climate and disaster risks to which the project is exposed to (E.g. \[document or free text to list and explain risks\]). This tends to be part of the appraisal documents..
-    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document), set `.documentType` to ''climateAndDisasterRiskAssessment'' and map any free text to `.description`.'
+    mapping: 'Project level: [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to ''climateAndDisasterRiskAssessment'' and map any free text to its [`.description`](project-schema.json,/definitions/Document,description).'
```
```suggestion
    mapping: |
        Project level: 
        
        [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'climateAndDisasterRiskAssessment' and map any free text to its [`.description`](project-schema.json,/definitions/Document,description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -482,17 +479,17 @@
         Project-level:
 
-        1. Add a `Metric` object to the `.forecasts` array.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.forecasts`](project-schema.json,,forecasts) array.
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`forecasts`](project-schema.json,,forecasts) array.
```

Also above (but I can't directly comment on it) "Project-level" should be "Project level"

duncandewhurst replied:
Fixed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -572,19 +569,16 @@
         Project level:
 
-        1. Add a `CostMeasurement` object to the `.costMeasurements` array and set its:
+        1. Add a [`CostMeasurement`](../../reference/schema.md#costmeasurement) object to the [`.costMeasurements`](project-schema.json,,costMeasurements) array and set its [`.id`](project-schema.json,/definitions/CostMeasurement,id) incrementally.
```
```suggestion
        1. Add a [`CostMeasurement`](../../reference/schema.md#costmeasurement) object to the [`costMeasurements`](project-schema.json,,costMeasurements) array and set its [`.id`](project-schema.json,/definitions/CostMeasurement,id) incrementally.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -638,19 +633,19 @@
     mapping: |-
-        1. Get the `Finance` object in `.budget.finance` or `.contractingProcesses.summary.finance` that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
-        2. Set the the financing arrangement's properties according to the instrument type:
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```
```suggestion
        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -750,8 +743,8 @@
     mapping: |-
-        1. Get the `Finance` object in `.budget.finance` or `.contractingProcesses.summary.finance` that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
-        2. Map to the financing arrangement's `.value`.
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```
```suggestion
        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`budget.finance`](project-schema.json,,budget/finance) or [`contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -772,8 +765,8 @@
     mapping: |-
-        1. Get the `Finance` object in `.budget.finance` or `.contractingProcesses.summary.finance` that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
-        2. Map to the financing arrangement's `.source`.
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```
```suggestion
        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`budget.finance`](project-schema.json,,budget/finance) or [`contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -791,9 +784,9 @@
-        1. Get the `Finance` object in `.budget.finance` or `.contractingProcesses.summary.finance` that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
-        2. Get the `Organization` object in `.parties` that represents the party providing the finance. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its `.roles`.
-        3. Set the financing arrangement's `.financingParty` to the `.id` and `.name` of the organization.
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```
```suggestion
        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`budget.finance`](project-schema.json,,budget/finance) or [`contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -791,9 +784,9 @@
-        2. Get the `Organization` object in `.parties` that represents the party providing the finance. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its `.roles`.
-        3. Set the financing arrangement's `.financingParty` to the `.id` and `.name` of the organization.
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
+        2. Get the [`Organization`](../../reference/schema.md#organization) object in [`.parties`](project-schema.json,,parties) that represents the party providing the finance. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its [`.roles`](project-schema.json,/definitions/Organization,roles).
```
```suggestion
        2. Get the [`Organization`](../../reference/schema.md#organization) object in [`parties`](project-schema.json,,parties) that represents the party providing the finance. If none exists yet, [add an organization](../common.md#add-an-organization) and add 'funder' to its [`.roles`](project-schema.json,/definitions/Organization,roles).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -859,16 +852,16 @@
     indicator: Accessibility/ efficiency
     disclosure format: Disclose dates for project preparation \[start date, end date\]
-    mapping: 'Project level: Map to `preparationPeriod`.'
+    mapping: 'Project level: Map to [`preparationPeriod`](project-schema.json,,preparationPeriod).'
```
```suggestion
    mapping: |
        Project level: 
        
        Map to [`preparationPeriod`](project-schema.json,,preparationPeriod).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -886,12 +879,14 @@
+        - [`.type`](project-schema.json,/definitions/Milestone,type) to 'financing'
+        - [`.status`](project-schema.json,/definitions/Milestone,status) to 'met'
+
+        2. If disclosing the submission date, set the milestone's [`.title`](project-schema.json,/definitions/Milestone,title) to "Climate finance submission" and its [`.dateMet`](project-schema.json,/definitions/Milestone,dateMet) to the date of the submission
```
```suggestion
        2. If disclosing the submission date, set the milestone's [`.title`](project-schema.json,/definitions/Milestone,title) to "Climate finance submission" and its [`.dateMet`](project-schema.json,/definitions/Milestone,dateMet) to the date of the submission.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -951,15 +946,15 @@
         Commitment fee (per annum)
     mapping: |-
-        1. Get the `Finance` object in `.budget.finance` or `.contractingProcesses.summary.finance` that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
+        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`.budget.finance`](project-schema.json,,budget/finance) or [`.contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```
```suggestion
        1. Get the [`Finance`](../../reference/schema.md#finance) object in [`budget.finance`](project-schema.json,,budget/finance) or [`contractingProcesses.summary.finance`](project-schema.json,/definitions/ContractingProcessSummary,finance) that represents the financing arrangement. If none exists yet, [add a financing arrangement](../common.md#add-a-financing-arrangement).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
-          - `.dueDate` to the date on which the disbursement is planned to occur
-          - `.type` to 'payment'
-          - `.value` to the amount and currency of the planned disbursement
+        1. If the disbursement relates to a contracting processes, for example a payment from a funder to a supplier or to a subcontractor of a supplier, get the [`ContractingProcess`](../../reference/schema.md#contractingprocess) in the [`.contractingProcesses`](project-schema.json,,contractingProcesses) array to which the disbursement relates and add a [`Milestone`](../../reference/schema.md#mi ...
```
```suggestion
        1. If the disbursement relates to a contracting processes, for example a payment from a funder to a supplier or to a subcontractor of a supplier, get the [`ContractingProcess`](../../reference/schema.md#contractingprocess) in the [`contractingProcesses`](project-schema.json,,contractingProcesses) array to which the disbursement relates and add a [`Milestone`](../../reference/schema.md#milestone) object to its [`.summary.milestones`](project-schema.json,/definitions/ContractingProcessSummary,milestones) array. 
        Otherwise, if the disbursement relates to the project, for example a payment from a funder to the public authority, add a [`Milestone`](../../reference/schema.md#milestone) object to the project-level [`milestones`](project-schema.json,,milestones) array.
```

odscjen replied:
I just think its easier to read if the "Otherwise..." is in a new line

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
         For each actual disbursement:
 
-        - If the disbursement relates to a contracting processes, for example a payment from a funder to a supplier or subcontractor of a supplier, get the `ContractingProcess` in the `.contractingProcesses` array to which the disbursement relates and add a `Transaction` object to its `.summary.transactions` array. Otherwise, if the disbursement relates to the project, for example a payment from  ...
+        1. If the disbursement relates to a contracting processes, for example a payment from a funder to a supplier or to a subcontractor of a supplier, get the [`ContractingProcess`](../../reference/schema.md#contractingprocess) in the [`.contractingProcesses`](project-schema.json,,contractingProcesses) array to which the disbursement relates and add a [`Transaction`](../../reference/schema.md# ...
```
```suggestion
        1. If the disbursement relates to a contracting processes, for example a payment from a funder to a supplier or to a subcontractor of a supplier, get the [`ContractingProcess`](../../reference/schema.md#contractingprocess) in the [`contractingProcesses`](project-schema.json,,contractingProcesses) array to which the disbursement relates and add a [`Transaction`](../../reference/schema.md#transaction) object to its [`.summary.transactions`](project-schema.json,/definitions/ContractingProcessSummary,transactions) array. 
        Otherwise, if the disbursement relates to the project, for example a payment from a funder to the public authority, add a [`Transaction`](../../reference/schema.md#transaction) object to the project-level [`transactions`](project-schema.json,,transactions) array.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
+        - [`.value`](project-schema.json,/definitions/Transaction,value) to the amount and currency of the disbursement
 
-        - Get the `Organization` in `.parties` that represents the payer. If none exists yet, [add an organization](../common.md#add-an-organization) for the payer:
+        3. Get the [`Organization`](../../reference/schema.md#organization) in [`.parties`](project-schema.json,,parties) that represents the payer. If none exists yet, [add an organization](../common.md#add-an-organization) for the payer:
```
```suggestion
        3. Get the [`Organization`](../../reference/schema.md#organization) in [`parties`](project-schema.json,,parties) that represents the payer. If none exists yet, [add an organization](../common.md#add-an-organization) for the payer:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
-          - Add 'payer' to the organization's `.roles` array
-          - Set the transaction's `.payer` to the `.id` and `.name` of the organization
+        - Add 'payer' to the organization's [`.roles`](project-schema.json,/definitions/Organization,roles) array
+        - Set the transaction's [`.payer`](project-schema.json,/definitions/Transaction,payer) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization
```
```suggestion
        - Set the transaction's [`.payer`](project-schema.json,/definitions/Transaction,payer) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
-          - `.value` to the amount and currency of the disbursement
+        - [`.id`](project-schema.json,/definitions/Transaction,id) incrementally
+        - [`.date`](project-schema.json,/definitions/Transaction,date) to the date of the disbursement
+        - [`.value`](project-schema.json,/definitions/Transaction,value) to the amount and currency of the disbursement
```
```suggestion
        - [`.value`](project-schema.json,/definitions/Transaction,value) to the amount and currency of the disbursement.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
+        - Set the transaction's [`.payer`](project-schema.json,/definitions/Transaction,payer) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization
 
-        - Get the `Organization` in `.parties` that represents the payee. If none exists yet, [add an organization](../common.md#add-an-organization) for the payee:
+        4. Get the [`Organization`](../../reference/schema.md#organization) in [`.parties`](project-schema.json,,parties) that represents the payee. If none exists yet, [add an organization](../common.md#add-an-organization) for the payee:
```
```suggestion
        4. Get the [`Organization`](../../reference/schema.md#organization) in [`parties`](project-schema.json,,parties) that represents the payee. If none exists yet, [add an organization](../common.md#add-an-organization) for the payee:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
-          - Add 'payee' to the organization's `.roles` array.
-          - Set the transaction's `.payee` to the `.id` and `.name` of the organization
+        - Add 'payee' to the organization's [`.roles`](project-schema.json,/definitions/Organization,roles) array.
+        - Set the transaction's [`.payee`](project-schema.json,/definitions/Transaction,payee) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization
```
```suggestion
        - Set the transaction's [`.payee`](project-schema.json,/definitions/Transaction,payee) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
+        - Set the transaction's [`.payee`](project-schema.json,/definitions/Transaction,payee) to the [`.id`](project-schema.json,/definitions/Organization,id) and [`.name`](project-schema.json,/definitions/Organization,name) of the organization
 
-        - Get the `Milestone` in `.milestones` that represents that planned disbursement:
+        5. Get the [`Milestone`](../../reference/schema.md#milestone) in [`.milestones`](project-schema.json,,milestones) that represents that planned disbursement:
```
```suggestion
        5. Get the [`Milestone`](../../reference/schema.md#milestone) in [`milestones`](project-schema.json,,milestones) that represents that planned disbursement:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1040,39 +1035,40 @@
-          - Set the transaction's `.relatedImplementationMilestone` to the `.id` and `.title` of the milestone
+        - Set its [`.status`](project-schema.json,/definitions/Milestone,status) to 'met'
+        - Set its [`.dateMet`](project-schema.json,/definitions/Milestone,dateMet) to the date of the disbursement
+        - Set the transaction's [`.relatedImplementationMilestone`](project-schema.json,/definitions/Transaction,relatedImplementationMilestone) to the [`.id`](project-schema.json,/definitions/Milestone,id) and [`.title`](project-schema.json,/definitions/Milestone,title) of the milestone
```
```suggestion
        - Set the transaction's [`.relatedImplementationMilestone`](project-schema.json,/definitions/Transaction,relatedImplementationMilestone) to the [`.id`](project-schema.json,/definitions/Milestone,id) and [`.title`](project-schema.json,/definitions/Milestone,title) of the milestone.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1230,27 +1221,27 @@
         Project-level:
 
-        1. Add a `Metric` object to the `.metrics` array.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.metrics`](project-schema.json,,metrics) array.
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`metrics`](project-schema.json,,metrics) array.
```

plus another "Project-level" which should be "Project level" above

duncandewhurst replied:
Fixed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1285,15 +1276,15 @@
         For each set of decommissioned assets in a specified location, create a new OC4IDS project and:
 
-        1. Set `.type` to 'decommissioning'
+        1. Set its [`.type`](project-schema.json,,type) to 'decommissioning'
```
```suggestion
        1. Set its [`type`](project-schema.json,,type) to 'decommissioning'.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1285,15 +1276,15 @@
+        1. Set its [`.type`](project-schema.json,,type) to 'decommissioning'
 
-        2. Set `.title` to the name of the asset
+        2. Set its [`.title`](project-schema.json,,title) to the name of the asset
```
```suggestion
        2. Set its [`title`](project-schema.json,,title) to the name of the asset.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1285,15 +1276,15 @@
+        2. Set its [`.title`](project-schema.json,,title) to the name of the asset
 
-        3. Add a `RelatedProject` object to the `.relatedProjects` array and set its:
+        3. Add a [`RelatedProject`](../../reference/schema.md#relatedproject) object to the [`.relatedProjects`](project-schema.json,,relatedProjects) array and set its:
```
```suggestion
        3. Add a [`RelatedProject`](../../reference/schema.md#relatedproject) object to the [`relatedProjects`](project-schema.json,,relatedProjects) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1336,17 +1327,17 @@
         Project-level:
 
-        1. Add a `Metric` object to the `.forecasts` array.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.forecasts`](project-schema.json,,forecasts) array.
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`forecasts`](project-schema.json,,forecasts) array.
```

plus "Project-level" should be "Project level" above

duncandewhurst replied:
Fixed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1438,9 +1426,9 @@
         Project-level:
 
-        1. [Add a project document](../common.md#add-a-project-document), set `.documentType` to 'inclusiveDesign' and map any free text to `.description`.
+        1. [Add a project document](../common.md#add-a-project-document), set its [`.documentType`](project-schema.json,/definitions/Document,documentType) to 'inclusiveDesign' and map any free text to its [`.description`](project-schema.json,/definitions/Document,description).
```
"Project-level" should be "Project level" above

duncandewhurst replied:
Fixed in 73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1468,10 +1456,10 @@
-        1. Set `.social.inIndigenousLand` to `true`
-        2. Add a `Location` object to the `.locations` array, set its `.id` incrementally and set its description to "Indigenous land: <Name> (<Category>)" substituting <Name>  and <Category> for the name and land category from the Landmark database.
+        1. Set [`.social.inIndigenousLand`](project-schema.json,/definitions/Social,inIndigenousLand) to `true`
+        2. Add a [`Location`](../../reference/schema.md#location) object to the [`.locations`](project-schema.json,,locations) array, set its [`.id`](project-schema.json,/definitions/Location,id) incrementally and set its description to "Indigenous land: <Name> (<Category>)" substituting <Name> and <Category> for the name and land category from the Landmark database.
```
```suggestion
        2. Add a [`Location`](../../reference/schema.md#location) object to the [`locations`](project-schema.json,,locations) array, set its [`.id`](project-schema.json,/definitions/Location,id) incrementally and set its description to "Indigenous land: <Name> (<Category>)" substituting <Name> and <Category> for the name and land category from the Landmark database.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1576,14 +1564,14 @@
-        1. For each labor obligation in the contract, add a code from the [laborObligations](../../reference/codelists.md#laborobligations) codelist to the`.summary.social.laborObligations.obligations` array.
-        2. Optionally, add a further explanation of the labor obligations to `.summary.social.laborObligations.description`.
+           1. For each labor obligation, add a code from the open [laborObligations](../../reference/codelists.md#laborobligations) codelist to the [`.summary.social.laborObligations.obligations`](project-schema.json,/definitions/ContractingProcessSummary,social/laborObligations/obligations) array.
+           2. Optionally, add a further explanation of the labor obligations to [`.summary.social.laborObligations.description`](project-schema.json,/definitions/ContractingProcessSummary,social/laborObligations/description).
```
```suggestion
           2. Optionally, add a further explanation of the labor obligations to [`summary.social.laborObligations.description`](project-schema.json,/definitions/ContractingProcessSummary,social/laborObligations/description).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1647,13 +1635,27 @@
 
-        1. Add a `Metric` object to the `.metrics` array, set its `.id` incrementally and set its `.title` to "Worker accidents".
-        2. For each reporting period, add an `Observation` object to the metric's `.observations` array, set its `.id` incrementally, set its `.period` to the reporting period and set its `.measure` to the number of accidents. If further details available, add them to the observation's `.notes` field.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Worker accidents".
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Worker accidents".
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1647,13 +1635,27 @@
 
-        1. Add a `Metric` object to the `.metrics` array, set its `.id` incrementally and set its `.title` to "Worker fatalities".
-        2. For each reporting period, add an `Observation` object to the metric's `.observations` array, set its `.id` incrementally, set its `.period` to the reporting period and set its `.measure` to the number of fatalities. If further details available, add them to the observation's `.notes` field.
+        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`.metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Worker fatalities".
```
```suggestion
        1. Add a [`Metric`](../../reference/schema.md#metric) object to the [`metrics`](project-schema.json,,metrics) array, set its [`.id`](project-schema.json,/definitions/Metric,id) incrementally and set its [`.title`](project-schema.json,/definitions/Metric,title) to "Worker fatalities".
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1740,10 +1744,11 @@
 
-        1. For each material test, add a code from the [constructionMaterial](../../reference/codelists.md#constructionmaterial) codelist to the`.social.healthAndSafety.materialTests.tests` array.
-        2. Add any further explanation of the tests to `.social.healthAndSafety.materialTests.description` including the international or national standards the tests conform to.
+        1. For each material test, add a code from the open [constructionMaterial](../../reference/codelists.md#constructionmaterial) codelist to the [`.social.healthAndSafety.materialTests.tests`](project-schema.json,/definitions/HealthAndSafety,materialTests/tests) array.
```
```suggestion
        1. For each material test, add a code from the open [constructionMaterial](../../reference/codelists.md#constructionmaterial) codelist to the [`social.healthAndSafety.materialTests.tests`](project-schema.json,/definitions/HealthAndSafety,materialTests/tests) array.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1740,10 +1744,11 @@
-        2. Add any further explanation of the tests to `.social.healthAndSafety.materialTests.description` including the international or national standards the tests conform to.
+        1. For each material test, add a code from the open [constructionMaterial](../../reference/codelists.md#constructionmaterial) codelist to the [`.social.healthAndSafety.materialTests.tests`](project-schema.json,/definitions/HealthAndSafety,materialTests/tests) array.
+
+        2. Add any further explanation of the tests to [`.social.healthAndSafety.materialTests.tests`](project-schema.json,/definitions/HealthAndSafety,materialTests/tests), including the international or national standards that the tests conform to.
```
```suggestion
        2. Add any further explanation of the tests to [`social.healthAndSafety.materialTests.tests`](project-schema.json,/definitions/HealthAndSafety,materialTests/tests), including the international or national standards that the tests conform to.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1791,27 +1796,25 @@
-        - relationship: indirect, stage: implementation
-        - relationship: direct, stage: operation
-        - relationship: indirect, stage: operation
+        2. For each job relationship - stage pair, add an [`Observation`](../../reference/schema.md#observation) object and set its:
```
```suggestion
        2. For each job relationship - stage pair, add an [`Observation`](../../reference/schema.md#observation) object to the metric's [`.observations`](project-schema.json,/definitions/Metric,observations) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1791,27 +1796,25 @@
+        - [`.id`](project-schema.json,/definitions/Observation,id) incrementally
+        - [`.dimensions.relationship`](project-schema.json,/definitions/Observation,dimensions) to the relationship ("direct" or "indirect")
+        - [`.dimensions.stage`](project-schema.json,/definitions/Observation,dimensions) to the stage ("implementation" or "operation")
+        - [`.measure`](project-schema.json,/definitions/Observation,measure) to the number of jobs
```
```suggestion
        - [`.measure`](project-schema.json,/definitions/Observation,measure) to the number of jobs.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1791,27 +1796,25 @@
-        - relationship: indirect, stage: implementation
-        - relationship: direct, stage: operation
-        - relationship: indirect, stage: operation
+        2. For each job relationship - stage pair, add an [`Observation`](../../reference/schema.md#observation) object and set its:
```
```suggestion
        2. For each job relationship - stage pair, add an [`Observation`](../../reference/schema.md#observation) object to the metric's [`.observations`](project-schema.json,/definitions/Metric,observations) array and set its:
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1916,13 +1919,21 @@
+
+        - [`.scheme`](project-schema.json,/definitions/Classification,scheme) to 'sdg'
+        - [`.id`](project-schema.json,/definitions/Classification,id) to the goal's number
+        - [`.description`](project-schema.json,/definitions/Classification,description) to the goal's title
```
```suggestion
        - [`.description`](project-schema.json,/definitions/Classification,description) to the goal's title.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1916,13 +1919,21 @@
-        For each United Nations Sustainable Development Goal Target to which the project is aligned, add a `Classification` object to the `.additionalClassifications` array, set its `.scheme` to 'sdgTarget', set its `.id` to the target's number and set its `.description` to the target's title.
+        - [`.scheme`](project-schema.json,/definitions/Classification,scheme) to 'sdgTarget'
+        - [`.id`](project-schema.json,/definitions/Classification,id) to the target's number
+        - [`.description`](project-schema.json,/definitions/Classification,description) to the target's title
```
```suggestion
        - [`.description`](project-schema.json,/definitions/Classification,description) to the target's title.
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -1999,17 +2004,22 @@
+        - [`.publicOffice.person.name`](project-schema.json,/definitions/PublicOffice,person/name) to the name of the person representing the public office present at the meeting
+        - [`.publicOffice.jobTitle`](project-schema.json,/definitions/PublicOffice,jobTitle) to the job title of the person representing the public office present at the meeting
+
+        2. Get the [`Organization`](../../reference/schema.md#organization) in [`.parties`](project-schema.json,,parties) that represents the public office. If none exists yet, [add an organization](../common.md#add-an-organization).
```
```suggestion
        2. Get the [`Organization`](../../reference/schema.md#organization) in [`parties`](project-schema.json,,parties) that represents the public office. If none exists yet, [add an organization](../common.md#add-an-organization).
```

odscjen commented on mapping/sustainability.yaml:
```diff
@@ -2050,14 +2060,17 @@
-          - `.id` incrementally
-          - `.name` to the beneficial owner's name
-          - `.identifier` to the beneficial owner's identifier
+        1. Get the [`Organization`](../../reference/schema.md#organization) in [`.parties`](project-schema.json,,parties) that represents the contractor or supplier.
```
```suggestion
        1. Get the [`Organization`](../../reference/schema.md#organization) in [`parties`](project-schema.json,,parties) that represents the contractor or supplier.
```

odscjen commented on docs/cost/common.md:
```diff
@@ -4,24 +4,26 @@ To avoid repetition in the mapping, we refer and link to the following common op
 
-* Add an `Organization` object to the `.parties` array and set its `.name` to the name of the organization.
-* If you collect organization identifiers, set `.identifier` according to the [identifier reference](../reference/schema.md#identifier) and set `.id` to {identifier.scheme}-{identifier.id}(-{department-identifier}). Otherwise, set `.id` incrementally.
+1. Add an `Organization` object to the `.parties` array and set its `.name` to the name of the organization.
```
```suggestion
1. Add an `Organization` object to the `parties` array and set its `.name` to the name of the organization.
```

odscjen reviewed (changes requested):
It looks like lots but they're mostly very minor copy edits in sustainability.yaml and common.md and they mostly fall into the following main categories:
1. adding a full stop at the end of a step.
2. rearranging the mapping so that it always starts with the stage (unless it can be either) and then a line between that and the actual instructions, I added those in for visual consistency.
3. removing full stops from in front of top level fields, e.g. `.parties` should be `parties`, or where the Contracting process has already been specified as the level then `.summary` changed to `summary`.

A couple of other things I noticed:
* There's a few places where it says "Project-level" instead of "Project level" but I couldn't put it in as a suggestion as it's not from this PR. I think I noted them all but I might have missed some.
* Lots of the "disclosure format" cells that are pulling in from the CoST spreadsheet are missing bullet points so they don't look like proper lists, e.g. Institutional policy coherence.

duncandewhurst committed:
Apply suggestions from code review

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

duncandewhurst committed:
docs/cost/common.md: Fix formatting, link fields to schema reference

duncandewhurst committed:
mapping/sustainability.yaml: Action review comments, fix block chomping

duncandewhurst committed:
mapping/sustainability.yaml: Fix sustainability element formatting, lint

duncandewhurst commented:
> * There's a few places where it says "Project-level" instead of "Project level" but I couldn't put it in as a suggestion as it's not from this PR. I think I noted them all but I might have missed some.

Fixed in https://github.com/open-contracting/infrastructure/pull/459/commits/73bc7abfdb6a4bf562298f1f80ac2d6cfd7656ec

>  * Lots of the "disclosure format" cells that are pulling in from the CoST spreadsheet are missing bullet points so they don't look like proper lists, e.g. Institutional policy coherence.

Fixed in https://github.com/open-contracting/infrastructure/pull/459/commits/e3babaaf60d2d8243f357d52f2886185a0a72b5f

duncandewhurst committed:
docs/cost: Use csv-table directive for mapping tables

duncandewhurst commented:
@jpmckinney a few suggestions for your review:

[512eb18](https://github.com/open-contracting/infrastructure/commit/512eb18d4d23dc6760470a3f9831dd5a2e4b6d19) adds to `manage.py` the script that I used to link field names to the schema reference tables. I thought that it might come in handy as it could be run against other Markdown content in the standard, but I'm happy to remove it if it's not going to be useful or if it's too much work to get the code up to the necessary standard.

There are two commits with large diffs that may be best to review separately:

* https://github.com/open-contracting/infrastructure/pull/459/commits/4fbfb325c8a636424322e041066b9db6bbd24c91 reorders the top-level fields
* https://github.com/open-contracting/infrastructure/pull/459/commits/10f403eb3e48878715ebc212dc909b171f3054a6 links field names in `sustainability.yaml` to the schema reference tables (no worries if you're not concerned about reviewing `sustainability.yaml`- @odscjen did a detailed review already).

jpmckinney committed:
build: Merge remove_column() into setup()

jpmckinney committed:
build: Add jsonschema to requirements

jpmckinney committed:
Run ./manage.py update

- Add Extension column to Finance codelists
- Correct encoding of currency
- Small modifications to documentType, milestoneCode, milestoneType codes
- Small modifications to schema descriptions
- Change schema property order

jpmckinney committed:
build: Rewrite manage.py so that methods are nearer where they are called

jpmckinney committed:
docs: Tidy changelog (mostly order)

jpmckinney reviewed (commented):
Please review the changelog:

* The changelog for #450 uses "deprecate" but I think it was simply renamed (which is fine as we're not 1.0).
* On that note, I'm not sure that we need to deprecate 'completed' – we can just remove it (i.e. rename it).
* Some new fields seem to be missing, e.g. `policyAlignment`. I haven't checked codes.
* We can probably add some text at the top describing the major change, but we can also do this in a new PR.

I ended up doing a major refactor of manage.py. The `update` and `pre-commit` commands still work (they are the only ones I know how to use). I didn't touch the two update-sustainability commands. You'll have to test lint (there are no ambiguous unlinked fields to test against).

Note: You need to run `./manage.py update --ppp-base-url https://raw.githubusercontent.com/open-contracting-extensions/public-private-partnerships/1.0-dev/schema/patched/` when using the `update` command.

Otherwise, this PR looks good!

**Edit:** One of my commits re-ran `update` which did make minor changes.

jpmckinney committed:
build: Move lineterminator argument

jpmckinney commented:
Ah, milestoneType has some unwanted edits in c83ff80

duncandewhurst committed:
schema/project-level/codelists/projectStatus.csv: Remove 'completed'

duncandewhurst committed:
docs/reference/changelog.md: Fix `budget.Breakdowns` entry

duncandewhurst committed:
docs/reference/changelog.md: Add missing, reorder

duncandewhurst committed:
docs/reference/changelog.md: Add missing codelists and codes

duncandewhurst committed:
manage.py: Fix lint command

duncandewhurst committed:
schema/project-level/codelists/milestoneType.csv: Revert unwanted edits

duncandewhurst committed:
schema/project-level/codelists/projectStatus.csv: Fix test errors

duncandewhurst commented:
Thanks for the changelog tidy and `manage.py` refactor! All of your changes look good to me.

>     * The changelog for [project-schema.json: deprecate budget.budgetBreakdown, add budget.bud… #450](https://github.com/open-contracting/infrastructure/pull/450) uses "deprecate" but I think it was simply renamed (which is fine as we're not 1.0).

Done in b1ed5783fa2f89f054ebd5719493848f01082691. I used 'replace' rather than 'rename' since the field's items also changed.

>     * On that note, I'm not sure that we need to deprecate 'completed' – we can just remove it (i.e. rename it).

Done in d36e14202775b4c8cfe28ef713369c683461b1da

>     * Some new fields seem to be missing, e.g. `policyAlignment`. I haven't checked codes.

Fixed in 4d9b86c6780adaf790911f5696b12d776213b305. I checked the new fields and I think only `policyAlignment` was missing. I also moved the entry for `budget.description` to the list of added fields.

I've checked codelists and codes and added the missing ones in 6f7554d62092c9b6fdf22f6a7ef62be6862b37f9

>     * We can probably add some text at the top describing the major change, but we can also do this in a new PR

A new PR sounds good.

> I ended up doing a major refactor of manage.py. The `update` and `pre-commit` commands still work (they are the only ones I know how to use). I didn't touch the two update-sustainability commands. You'll have to test lint (there are no ambiguous unlinked fields to test against).

Confirmed that the `update-sustainability` commands still work. `lint -l` was broken, but I've fixed it in dc86a522add6a8548dddc11871371462fee4bc60.

> **Edit:** One of my commits re-ran `update` which did make minor changes.

Changes look good, aside from:

> Ah, milestoneType has some unwanted edits in [c83ff80](https://github.com/open-contracting/infrastructure/commit/c83ff80d30e9a4baae93c78dd4c0b3cd2497c484)

Reverted in https://github.com/open-contracting/infrastructure/pull/459/commits/08ed35f6c98d1ae26e8ad4bcf8dfb333f9382f79

jpmckinney committed:
chore: Use tuple() earlier

jpmckinney committed:
changelog: Alphabetical order for codes, codelists
